polar-core/src/polar.rs

```diff
@@ -398,4 +395,65 @@ mod tests {
         );
         assert!(!polar.kb.read().unwrap().has_rules());
     }
+
+    #[test]
+    fn test_valid_shorthand_rules_rewritten_in_presence_of_other_issues() {
+        let polar = Polar::new();
+        let src = r#"
+            allow(actor, action, resource) if
```

Comment:
Not sure I totally understand this test — what’s the error that would have prevented these rules from being rewritten into the KB?

Reply:
Ah yeah that's a good shout — it's not totally obvious. I'll add a comment explaining it

Reply:
Simpler test in af960f69